sdk/cmake/host-tools.cmake

```diff
@@ -29,6 +29,48 @@ function(setup_host_tools)
         set(HOST_MODULE_SUFFIX ".so")
     endif()
 
+    # Normalize to the same format as our own ARCH, and add one for the VC shell
+    string(TOLOWER "${CMAKE_HOST_SYSTEM_PROCESSOR}" lowercase_CMAKE_HOST_SYSTEM_PROCESSOR)
+    if(lowercase_CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL x86 OR lowercase_CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "^i[3456]86$")
+        set(HOST_ARCH i386)
+        set(VCVARSALL_ARCH x86)
+    elseif(lowercase_CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL x86_64 OR lowercase_CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL amd64)
+        set(HOST_ARCH amd64)
+        set(VCVARSALL_ARCH amd64_x86) # x64 host-tools are not happy compiling for x86...
```

Comment:
For the record,
trying with `VCVARSALL_ARCH amd64`:

msvc-arm and msbuild-i386 are not affected.

msvc-i386 and clang-cl-i386 are affected:
cabman, at least, fails:
```
C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um\winbase.h(9564): error C2065: 'InterlockedIncrement64': undeclared identifier
C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um\winbase.h(9596): error C2065: 'InterlockedDecrement64': undeclared identifier
C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um\winbase.h(9632): error C3861: 'InterlockedExchange64': identifier not found
C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um\winbase.h(9686): error C3861: 'InterlockedExchangeAdd64': identifier not found
C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um\winbase.h(9696): error C3861: 'InterlockedExchangeAdd64': identifier not found
C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um\winbase.h(9743): error C3861: 'InterlockedAnd64': identifier not found
C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um\winbase.h(9753): error C3861: 'InterlockedOr64': identifier not found
C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um\winbase.h(9763): error C3861: 'InterlockedXor64': identifier not found
```
Any idea about the cause? MS or ROS?